tested/oracles/value.py

```diff
@@ -233,9 +233,10 @@ def _check_data_type(
             prepared_actual.data if isinstance(prepared_actual.data, list) else []
         )
         prepared_elements = []
-        for expected_element, actual_element in zip(expected_elements, actual_elements):
+        for i, expected_element in enumerate(expected_elements):
```

Comment:
Using this method, won't we miss if the actual map contains more elements than expected? 

I think we would also need a check that both have the same length